### PR TITLE
Product filtering (using checkboxes for categories & reset)

### PR DESCRIPTION
**Pull request by harrydigos**

### CODE REVIEW

File: components/products/ProductFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+const ProductFilter = ({ onFilter }: { onFilter: Dispatch<SetStateAction<string[]>> }) => {
+  const { register, handleSubmit, reset } = useForm<CheckFields>({
+    mode: "onChange",
+    defaultValues: { ...checkboxes.reduce((acc, curr) => ({ ...acc, [curr]: false }), {}) },
+  });
+
+  const onSuccessfulSubmit = (data: CheckFields) => {
+    let appliedFilters: string[] = [];
+    Object.entries(data).forEach(([key, value]) => (value ? appliedFilters.push(key) : null));
```
Comment (panteliselef) [inline components/products/ProductFilter.tsx]: use .filter() for this

File: components/products/ProductFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        <div className="checkmark" />
+        <div className="text" data-text={item} />
+      </label>
+    );
+  }
+);
+
+const checkboxes = ["Espresso", "Beans", "Pods", "Frothers", "Grinders", "Accessories", "Cups"] as const;
+
+type TupleToObject<T extends readonly string[]> = {
```
Comment (panteliselef) [inline components/products/ProductFilter.tsx]: this is a utility type, it'd be nice to have it a different file

File: components/products/ProductFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      <label className="checkbox-container">
+        <input {...props} ref={ref} type="checkbox" />
+        <div className="checkmark" />
+        <div className="text" data-text={item} />
+      </label>
+    );
+  }
+);
+
+const checkboxes = ["Espresso", "Beans", "Pods", "Frothers", "Grinders", "Accessories", "Cups"] as const;
```
Comment (panteliselef) [inline components/products/ProductFilter.tsx]: @BabisDigos  ❓why are those capitalised ?

**Review by panteliselef — CHANGES_REQUESTED** [review]

File: components/products/ProductFilter.tsx (hunk 3 above)
Comment (harrydigos) [reply components/products/ProductFilter.tsx]: In DB I have them like that, and I was bored changing them.

File: components/products/ProductFilter.tsx (hunk 1 above)
Comment (harrydigos) [reply components/products/ProductFilter.tsx]: Tried it but it didn't work as I wanted

**Review by panteliselef — APPROVED** [review]

**Review by panteliselef — APPROVED** [review]
💯

**Pull request merged**